api/raml/core-metadata.raml

```diff
@@ -546,10 +546,12 @@ schemas:
     delete:
         description: Remove the DeviceProfile designated by database generated id. This does not remove associated commands. Returns Internal Service Error (HTTP 500) for unknown or unanticipated issues. Returns NotFoundException (HTTP 404) if the device profile cannot be found by the identifier provided. Returns DataValidationException (HTTP 413) if devices still reference the profile.
         responses:
-            "200":
-                description: boolean indicating success of the remove operation
+            "204":
+                description: successful deletion of the device profile
             "404":
                 description: if the device profile cannot be found with the identifier provided
+            "409":
+                description: if devices or provision watchers still reference the profile
```

Comment:
This change IS backward compatible because it's additive. It didn't exist at all in Edinburgh and has been added for Fuji so an Edinburgh client will continue to work.